### PR TITLE
msbuild-structured-log-viewer: init at 2.2.356

### PR DESCRIPTION
**Pull request by js6pak**
An Avalonia-based viewer for the MSBuild binary log format, which allows you to inspect basically every aspect of a .NET build
https://github.com/KirillOsenkov/MSBuildStructuredLog

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ms/msbuild-structured-log-viewer/package.nix [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    icu
+    openssl
+    libkrb5
+  ];
+
+  dontDotnetFixup = true;
+
+  postFixup =
+    ''
+      wrapDotnetProgram $out/lib/${finalAttrs.pname}/StructuredLogViewer.Avalonia $out/bin/${meta.mainProgram}
```
Comment (js6pak) [inline pkgs/by-name/ms/msbuild-structured-log-viewer/package.nix:52]: I'm renaming the executable name from upstream's `StructuredLogViewer.Avalonia` to `msbuild-structured-log-viewer` here, but is that something we should/can do?

File: pkgs/by-name/ms/msbuild-structured-log-viewer/package.nix [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    exec = meta.mainProgram;
+    categories = [ "Development" ];
+  };
+
+  passthru.updateScript = writeShellScript "update-${finalAttrs.pname}" ''
+    ${lib.getExe nix-update}
+    "$(nix-build -A "$UPDATE_NIX_ATTR_PATH.fetch-deps" --no-out-link)"
+  '';
+
+  meta = with lib; {
```
Comment (Daholli) [inline pkgs/by-name/ms/msbuild-structured-log-viewer/package.nix:81]:
```suggestion
  meta = {
```

File: pkgs/by-name/ms/msbuild-structured-log-viewer/package.nix [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    description = "A rich interactive log viewer app for MSBuild logs";
+    homepage = "https://github.com/KirillOsenkov/MSBuildStructuredLog";
+    changelog = "https://github.com/KirillOsenkov/MSBuildStructuredLog/releases/tag/v${finalAttrs.version}";
+    sourceProvenance = with sourceTypes; [
+      fromSource
+      binaryBytecode
+      binaryNativeCode
+    ];
+    license = licenses.mit;
```
Comment (Daholli) [inline pkgs/by-name/ms/msbuild-structured-log-viewer/package.nix:90]:
```suggestion
    license = lib.licenses.mit;
```

File: pkgs/by-name/ms/msbuild-structured-log-viewer/package.nix [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  passthru.updateScript = writeShellScript "update-${finalAttrs.pname}" ''
+    ${lib.getExe nix-update}
+    "$(nix-build -A "$UPDATE_NIX_ATTR_PATH.fetch-deps" --no-out-link)"
+  '';
+
+  meta = with lib; {
+    description = "A rich interactive log viewer app for MSBuild logs";
+    homepage = "https://github.com/KirillOsenkov/MSBuildStructuredLog";
+    changelog = "https://github.com/KirillOsenkov/MSBuildStructuredLog/releases/tag/v${finalAttrs.version}";
+    sourceProvenance = with sourceTypes; [
```
Comment (Daholli) [inline pkgs/by-name/ms/msbuild-structured-log-viewer/package.nix:85]:
```suggestion
    sourceProvenance = with lib.sourceTypes; [
```

File: pkgs/by-name/ms/msbuild-structured-log-viewer/package.nix [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      fromSource
+      binaryBytecode
+      binaryNativeCode
+    ];
+    license = licenses.mit;
+    maintainers = with maintainers; [ js6pak ];
```
Comment (Daholli) [inline pkgs/by-name/ms/msbuild-structured-log-viewer/package.nix:91]:
```suggestion
    maintainers = with lib.maintainers; [ js6pak ];
```

**Review by Daholli — CHANGES_REQUESTED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**